### PR TITLE
Disposal rate method

### LINKED ISSUE #931: [FEAT] Disposal Rate Method from Friedland FreqSev Approach #3
### Description

P207 of Friedland describes a disposal rate method that is not easily supported by the current `IncrementalAdditive` class. At the same time, this method doesn't generate a new IBNR estimate. Rather, it re-apportions the period-by-period emergence of the total IBNR that's already given. This would appear to fall under the Adjustment bucket of classes. Therefore, we have 3 approaches for implementing this method

- add a new DisposalRate development method. it would largely mimic the incremental additive method, with some additional calculations 
- add a new parameter to the existing incremental additive method to specify whether it's performing the Schmidt variant or the Disposal rate variant. 
- add a new DisposalRate adjustment class to handle the additional calculations, since it doesn't estimate a new IBNR

### Is your feature request aligned with the scope of the package?

- [x] Yes, absolutely!
- [ ] No, but it's still worth discussing.
- [ ] N/A (this request is not a codebase enhancement).

### Describe the solution you'd like, or your current workaround.

At the present I'm fairly indifferent which approach we take. Would like to hear everyone's thoughts

### Do you have any additional supporting notes?

_No response_

### PR DESCRIPTION
**Pull request by henrydingliu**
## Summary of Changes  
Implement the disposal rate method from Friedland FreqSev Approach 3 as an adjustment estimator 

## Related GitHub Issue(s)  
closes #931 

## Additional Context for Reviewers  
- Created a DisposalMixin for the new adjustment estimator and Triangle
    - Berquist-Sherman also uses disposal rate. will leave harmonization to another PR
    - We may eventually want to split Common into DevelopmentMixin, IncrementalMixin, etc. 
- Added disposal_rate_tri to Triangle, mimicking age_to_age/link_ratio
- Added is_disposal_rate to Triangle, in order to differentiate the proper path in incr_to_cum and cum_to_incr

- [x] I passed tests locally for both code (`uv run pytest`) and documentation changes (`uv run jb build docs --builder=custom --custom-builder=doctest`)

---

> [!NOTE]
> **Medium Risk**
> Changes core triangle cum/incr behavior for a new pattern type and introduces a reserving path that overwrites projected triangles from ultimates; mistakes could affect forecast outputs but scope is isolated behind disposal-rate flags and new estimator usage.
> 
> **Overview**
> Adds **Friedland disposal-rate** reserving as public `DisposalRate`, which fits a cumulative “% of ultimate” emergence pattern from a triangle and a required **`sample_weight` ultimate**, then projects the lower triangle via derived LDFs.
> 
> `Triangle` gains **`disposal_rate_tri`**, an **`is_disposal_rate`** flag, and **`DisposalMixin`** (`disposal_rate_` / `incr_disposal_rate_`) mixed into `TriangleBase`. **`incr_to_cum`** / **`cum_to_incr`** now treat disposal-rate patterns as additive cumulative ratios instead of multiplicative link-ratio patterns.
> 
> `MethodBase.validate_weight` is refactored to a **`@staticmethod`** for reuse from `DisposalRate`. Tests cover Friedland textbook fidelity, chainladder parity, sparse backends, and required-weight validation; **`DisposalRate`** is wired into the public API.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 210ec6790e1b546e9da6fff0b3fd372ed18780ad. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by genedan** [comment]
Just letting you guys know that I can look through this today.

File: chainladder/adjustments/tests/test_disposal.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            'ay': [2001,2002,2003,2004,2005,2006,2007,2008],
+            'dev':[2008,2008,2008,2008,2008,2008,2008,2008],
+        },
+        origin = 'ay',
+        development='dev',
+        columns='Closed Claim Counts',
+        cumulative=True,
+    )
+    dr = cl.DisposalRate(n_periods = 5, average = 'simple', drop_high = 1, drop_low = 1).fit_transform(X=tri,sample_weight=ult_tri)
+    assert np.all(abs(dr.disposal_.round(3).values.flatten() - [.200,.433,.585,.710,.791,.862,.882,.912,1.000] <=0.001))
```
Comment (genedan) [inline chainladder/adjustments/tests/test_disposal.py]: I wonder if we should be using a higher level of precision than what Friedland has. That is, have the expectation values at the precision of the `atol` fixture.

Since this test passes, I think the numbers are correct, so we'd just change the rounding to 4 places and the expectation values at 4 decimals.

Interestingly, Friedland doesn't round on page 253 and also doesn't show the precision, so it looks like the paper isn't 100% consistent in that regard.

File: chainladder/adjustments/tests/test_disposal.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    lhs = (dr.full_triangle_.cum_to_incr()-tri.cum_to_incr()).round(0).values.flatten()
+    rhs = np.array([
+                                                            77.,  
+                                                    24.,    70.,  
+                                            12.,    18.,    54.,  
+                                    46.,    13.,    19.,    57.,  
+                            52.,    45.,    13.,    19.,    56.,  
+                    76.,    49.,    43.,    12.,    18.,    54.,  
+            67.,    55.,    36.,    31.,    9.,     13.,    39.,  
+    140.,   91.,    75.,    49.,    42.,    12.,    18.,    53.
```
Comment (genedan) [inline chainladder/adjustments/tests/test_disposal.py]: Friedland has 43 incremental claims at AY 2008 at 72 months. I calculated (609 - 127) / (1 - .2) (.862 - .791) = 42.7775.

File: chainladder/adjustments/tests/test_disposal.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    ])
+    assert np.all(abs(lhs[~np.isnan(lhs)] - rhs <= 1))
```
Comment (genedan) [inline chainladder/adjustments/tests/test_disposal.py:41]: Would you be brave enough to set the margin of error at 0, assuming the previous comment wasn't due to a rounding issue?

File: chainladder/adjustments/disposal.py [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+        import chainladder as cl
+        import numpy as np
+
+    .. testcode::
+
+        clrd = cl.load_sample('clrd').sum()
+        ult = cl.Chainladder().fit(clrd['IncurLoss']).ultimate_
+        dr = cl.DisposalRate().fit_transform(clrd['CumPaidLoss'],sample_weight = ult)
+
+    Once we apply this adjustment method via a `fit_transform`, we can examin the emergence pattern via `disposal_rate_tri`.
```
Comment (genedan) [inline chainladder/adjustments/disposal.py]: Small typo in "examine"

File: chainladder/adjustments/disposal.py [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+        )
+        if hasattr(self.X_, "w_"):
+            self.w_ = tw.fit(X=self.disposal_rate_tri * self.X_.w_).w_.values
+        else:
+            self.w_ = tw.fit(X=self.disposal_rate_tri).w_.values
+        #calculate factors
+        super().fit(ult.values,self.X_.values,self.w_)
+        #keep attributes
+        self.disposal_ = self._param_property(self.disposal_rate_tri,self.params_.slope_[...,0][..., None, :])
+        self.disposal_ = concat((self.disposal_,(self.X_.latest_diagonal*0 + 1).iloc[:,:,0,:].rename("development", [9999])),axis=3)
```
Comment (genedan) [inline chainladder/adjustments/disposal.py]: Should we turn this `9999` value into a default option for the package?

**Comment by genedan** [comment]
Do you think we should incorporate some of the latter steps in the method into the package - such as the severity trend selection (even though Friedland just wound up doing a judgmental 5%), and the tail selection of the incremental paid severities?

Or is the package already capable of these things for the Friedland recreation?

File: chainladder/adjustments/tests/test_disposal.py (hunk 1 above)
Comment (henrydingliu) [reply chainladder/adjustments/tests/test_disposal.py]: i'll add a new assertion statement with round(,4).

File: chainladder/adjustments/tests/test_disposal.py (hunk 2 above)
Comment (henrydingliu) [reply chainladder/adjustments/tests/test_disposal.py]: fat finger. will fix

File: chainladder/adjustments/tests/test_disposal.py (hunk 3 above)
Comment (henrydingliu) [reply chainladder/adjustments/tests/test_disposal.py:41]: i'll take a look.

File: chainladder/adjustments/disposal.py (hunk 4 above)
Comment (henrydingliu) [reply chainladder/adjustments/disposal.py]: will fix

File: chainladder/adjustments/disposal.py (hunk 5 above)
Comment (henrydingliu) [reply chainladder/adjustments/disposal.py]: great idea. alternatively, should it just be inferred based on the current `ULT_VAL`?

**Comment by henrydingliu** [comment]
> Do you think we should incorporate some of the latter steps in the method into the package - such as the severity trend selection (even though Friedland just wound up doing a judgmental 5%), and the tail selection of the incremental paid severities?
> 
> Or is the package already capable of these things for the Friedland recreation?

My current approach is to build enough basic functionality within the package such that they can be manually linked together to recreate friedland. That's why I split out `TriangleWeight` in #941, to handle the different regression periods for the severity trend selections. Once you help me work out the kinks on this, the next blocker is #1024. (Was really hoping one of the younger contributors could help me out on that one; reading latex formulas for #952 gave me enough headaches for this calendar year). if i run into more blockers after that, i'll create more issues. 

i haven't thought too much about long term if all this piecemeal functionality should be combined into a new freqsev method. we can decide as a collective once we can see a completed chapter 11 recreation. maybe it'll be too much hacking (i tried hacking the severity regression periods but it got too ugly) and we'll want to do some more integration. but if it's like, iono, a dozen or so lines and it's all using the public API (no numpy hacking tri.values), it could be a cool example of how to be innovative with the package. right now we are building a lego model based on the instructions. but the magic comes from combining the pieces in weird and unexpected ways. i mean, that's also the whole end goal of having a well-typed, well-doc'd API.

there is one major refactor that i'm fairly certain i want to push for at some point, but can't yet clearly articulate the right requirements. and that's the entire votingchainladder and pipeline workflow folder of the package. within friedland, and also fairly common in actual practice, actuaries are using reported and paid to estimate the same loss ultimate or using reported and closed to estimate the same count ultimate. votingchainladder currently doesn't support weighing between reported triangle and paid triangle. pipeline currently doesn't support using reported ultimate to recast closed disposal rate. so i think for votingchainladder and pipeline to be more than window dressing, it needs to accommodate multi-triangle interaction. and then however we refactor votingchainladder and pipeline will likely result in some further harmonizing on all the dev transformers, adjustment transformers, and ibnr predictors. not sure if all this would be 1.0 work. we can decide after an initial friedland recreation.

File: chainladder/adjustments/tests/test_disposal.py (hunk 3 above)
Comment (henrydingliu) [reply chainladder/adjustments/tests/test_disposal.py:41]: >Interestingly, Friedland doesn't round on page 253 and also doesn't show the precision, so it looks like the paper isn't 100% consistent in that regard.

turns out friedland wasn't rounding on page 245 either.

File: chainladder/adjustments/disposal.py (hunk 5 above)
Comment (genedan) [reply chainladder/adjustments/disposal.py]: We could, I'll open an issue.

File: chainladder/adjustments/tests/test_disposal.py (hunk 1 above)
Comment (henrydingliu) [reply chainladder/adjustments/tests/test_disposal.py]: got to full matching at 3 decimals.

File: chainladder/adjustments/tests/test_disposal.py (hunk 3 above)
Comment (henrydingliu) [reply chainladder/adjustments/tests/test_disposal.py:41]: couldn't reduce the tolerance due to the 91 at 2008, 36 months. (609-127)/0.8*(0.585 - 0.433) = 91.58

**Review by genedan — APPROVED** [review]
